### ISSUE TITLE
[Issue] Cannot create a list in Editor

### ISSUE THREAD

**Issue by alcoranpaul** [FlaxEngine/FlaxEngine#3110]
**Issue description:**

I wanted to create a list of `JsonAssetReferences` for a custom `JsonAsset`, however when creating the list via the `+` icon. It throws a warning.

These are the two instances that I cannot make a list.
1. In a `JsonAsset`
  - Log warning
``` csharp
[ 00:03:37.065 ]: [Warning] Exception: Object of type 'FlaxEngine.JsonAssetReference`1[Game.SomeDataAgain]' cannot be converted to type 'System.Collections.Generic.List`1[FlaxEngine.JsonAssetReference`1[Game.SomeDataAgain]]'.
[ 00:03:37.065 ]: [Warning]    at System.RuntimeType.CheckValue(Object& value, Binder binder, CultureInfo culture, BindingFlags invokeAttr)
   at System.Reflection.RtFieldInfo.SetValue(Object obj, Object value, BindingFlags invokeAttr, Binder binder, CultureInfo culture)
   at System.Reflection.FieldInfo.SetValue(Object obj, Object value)
   at FlaxEditor.Scripting.ScriptMemberInfo.SetValue(Object obj, Object value) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\Scripting\ScriptType.cs:line 690
   at FlaxEditor.CustomEditors.ValueContainer.Set(ValueContainer instanceValues, Object value) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\Values\ValueContainer.cs:line 393
   at FlaxEditor.CustomEditors.CustomEditor.SynchronizeValue(Object value) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 334
   at FlaxEditor.CustomEditors.CustomEditor.RefreshInternal() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 364
   at FlaxEditor.CustomEditors.CustomEditor.RefreshRootChild() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 325
   at FlaxEditor.CustomEditors.CustomEditor.RefreshRoot() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 278
   at FlaxEditor.CustomEditors.SyncPointEditor.RefreshInternal() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\SyncPointEditor.cs:line 80
   at FlaxEditor.CustomEditors.CustomEditorPresenter.PresenterPanel.Update(Single deltaTime) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditorPresenter.cs:line 91

  ```
2. In the Scripts window of an Actor
  - Log warning
```csharp
[ 00:05:03.320 ]: [Warning] Exception: Object of type 'FlaxEngine.Vector3' cannot be converted to type 'System.Collections.Generic.List`1[FlaxEngine.Vector3]'.
[ 00:05:03.320 ]: [Warning]    at System.RuntimeType.CheckValue(Object& value, Binder binder, CultureInfo culture, BindingFlags invokeAttr)
   at System.Reflection.RtFieldInfo.SetValue(Object obj, Object value, BindingFlags invokeAttr, Binder binder, CultureInfo culture)
   at System.Reflection.FieldInfo.SetValue(Object obj, Object value)
   at FlaxEditor.Scripting.ScriptMemberInfo.SetValue(Object obj, Object value) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\Scripting\ScriptType.cs:line 690
   at FlaxEditor.CustomEditors.ValueContainer.Set(ValueContainer instanceValues, Object value) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\Values\ValueContainer.cs:line 393
   at FlaxEditor.CustomEditors.CustomEditor.SynchronizeValue(Object value) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 334
   at FlaxEditor.CustomEditors.CustomEditor.RefreshInternal() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 364
   at FlaxEditor.CustomEditors.CustomEditor.RefreshRootChild() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 325
   at FlaxEditor.CustomEditors.CustomEditor.RefreshRoot() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 278
   at FlaxEditor.CustomEditors.Dedicated.ScriptsEditor.Refresh() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\Dedicated\ScriptsEditor.cs:line 1046
   at FlaxEditor.CustomEditors.SyncPointEditor.RefreshInternal() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\SyncPointEditor.cs:line 80
   at FlaxEditor.CustomEditors.CustomEditor.RefreshRootChild() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 325
   at FlaxEditor.CustomEditors.CustomEditor.RefreshRoot() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditor.cs:line 278
   at FlaxEditor.CustomEditors.SyncPointEditor.RefreshInternal() in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\SyncPointEditor.cs:line 85
   at FlaxEditor.CustomEditors.CustomEditorPresenter.PresenterPanel.Update(Single deltaTime) in E:\Game Development\FlaxMaster_Clone\FlaxEngine\Source\Editor\CustomEditors\CustomEditorPresenter.cs:line 91

```

**Steps to reproduce:**
- `JsonAsset`
  1. Create a custom classes, one that has arbitrary data, and one that holds a list of that asset
  2. Create the two `JsonAsset`s
  3. On the `JsonAsset` that has a list, create a list via `+` button.
  4. See to it that it throws a warning
- Scripts Editor
  1. Create a script that has a variable list, for any arbitrary type, that is visible in the editor
  2. Attach that script to an actor
  3. Create a list via `+` button in the script
  4. See to it that it throws a warning

**Video**

https://github.com/user-attachments/assets/8d6d4104-22e9-4c48-87bc-9f244bdc4607

**Log file**

[Log_2024_12_14_18_59_52.txt](https://github.com/user-attachments/files/18138379/Log_2024_12_14_18_59_52.txt)

**Flax version:**
Master ||  1.9.6606

**Comment by Myterian** [comment]
Related #3081

**Comment by Tryibion** [comment]
This one should have been fixed by this PR <https://github.com/FlaxEngine/FlaxEngine/pull/3109>

**Comment by alcoranpaul** [comment]
Okay, it was fixed now by #3109. Thanks `Tryibion`!!
